File: Source/Aladdin.CAPI.SCard.Java/src/aladdin/capi/scard/IAppletFileFolder.java
package aladdin.capi.scard;
import java.io.*; 

///////////////////////////////////////////////////////////////////////////
// Каталог файловой системы апплета
///////////////////////////////////////////////////////////////////////////
public interface IAppletFileFolder extends IAppletFileObject
{
    // список объектов файловой системы
    short[] EnumerateFolders(); short[] EnumerateFiles(); 

    // создать объект файловой системы
    IAppletFileFolder createFolder(short name, FileObjectInfo info) throws IOException; 
    IAppletFile       createFile  (short name, FileObjectInfo info) throws IOException; 

    // открыть объект файловой системы
    IAppletFileFolder openFolder(short name) throws IOException; 
    IAppletFile       openFile  (short name) throws IOException;

    // удалить объект файловой системы
    void removeFolder(short name) throws IOException; 
    void removeFile  (short name) throws IOException;
}

File: Source/Aladdin.CAPI.SCard/IAppletFileFolder.cs
using System;

namespace Aladdin.CAPI.SCard
{
    ///////////////////////////////////////////////////////////////////////////
    // Каталог файловой системы апплета
    ///////////////////////////////////////////////////////////////////////////
    public interface IAppletFileFolder : IAppletFileObject
    {
        // список объектов файловой системы
        UInt16[] EnumerateFolders(); UInt16[] EnumerateFiles(); 

        // создать объект файловой системы
        IAppletFileFolder CreateFolder(UInt16 name, FileObjectInfo info); 
        IAppletFile       CreateFile  (UInt16 name, FileObjectInfo info); 

        // открыть объект файловой системы
        IAppletFileFolder OpenFolder(UInt16 name); IAppletFile OpenFile(UInt16 name);

        // удалить объект файловой системы
		void RemoveFolder(UInt16 name); void RemoveFile(UInt16 name);
    }
}
